### PR TITLE
Improve building of builder images and docker base image

### PR DESCRIPTION
**Pull request by malena-ebert-sonarsource**

### CODE REVIEW

File: .cirrus.yml [hunk 1]
```diff
@@ -1,73 +1,104 @@
...
+    git reset --hard $CIRRUS_CHANGE_IN_REPO
+  fi
+
+image_build_task:
+  only_if: $CIRRUS_BRANCH == "master" || $CIRRUS_PR != ""
   auto_cancellation: true
   gce_instance:
     image_project: ubuntu-os-cloud
-    image_family: ubuntu-minimal-2004-lts
+    image_family: ubuntu-minimal-2010
```
Comment (tobias-trabelsi-sonarsource) [inline .cirrus.yml:22]: Any particular reason to not go for 2004-lts? 2010 will be EOL in ~6 month

File: docker/Dockerfile [hunk 2]
```diff
@@ -0,0 +1,67 @@
...
+  && curl -sSL https://packages.atlassian.com/api/gpg/key/public | apt-key add - \
+  && curl -sSL https://dl-ssl.google.com/linux/linux_signing_key.pub | apt-key add - \
+  && curl -sSL https://adoptopenjdk.jfrog.io/adoptopenjdk/api/gpg/key/public | apt-key add -
+
+# Node JS
+RUN echo "registry = https://repox.jfrog.io/repox/api/npm/npm" >> ~/.npmrc
+
+# Helper scripts
+COPY --chown=sonarsource:sonarsource bin/ /home/sonarsource/.local/bin/
+ENV PATH="/home/sonarsource/.local/bin:${PATH}"
```
Comment (tobias-trabelsi-sonarsource) [inline docker/Dockerfile:39]: just a minor point: PATH as docker env can be overwritten, so maybe echo it to `/etc/environment`?
`RUN echo "PATH="${PATH}:/home/sonarsource/.local/bin"`

**Review by tobias-trabelsi-sonarsource — APPROVED** [review]
Just two comments. if you disagree it looks good to me :+1:

File: .cirrus.yml (hunk 1 above)
Comment (malena-ebert-sonarsource) [reply .cirrus.yml:22]: oh forget to revert it after testing something, thanks

**Review by tobias-trabelsi-sonarsource — APPROVED** [review]

**Pull request merged**